contracts/01-tamagotchi/src/lib.rs

```diff
@@ -1,19 +1,58 @@
 #![no_std]
 
+use gstd::{debug, exec, msg};
 #[allow(unused_imports)]
 use gstd::prelude::*;
+use tamagotchi_io::{Tamagotchi, TmgAction, TmgEvent};
+
+static mut TAMAGOTCHI: Option<Tamagotchi> = None;
 
 #[no_mangle]
 extern fn init() {
     // TODO: 5️⃣ Initialize the Tamagotchi program
+    let name: String = msg::load()
+        .expect("Can't decode the init message");
+
+    debug!("Program was initialized with message {:?}",
+        name);
+
+    let tamagotchi = Tamagotchi {
+        name: name.clone(),
```

Comment:
No need to clone the `name`, it is better to move it.
```suggestion
        name,
```